# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/use_counter_metrics.yaml
# Commit: 5a65d2960e0d
# Full Hash: 5a65d2960e0d088d5f9b7ec573e6970014b58687
# Author: Matthew Gaudet <mgaudet@mozilla.com>
# Date: 2025-05-16 09:19:19
# Regressor Bug: 1964451
# File Overlap Count: 1
# Description:
#   Bug 1964451 - Add use counters for OOM handling r=sfink,mccr8
#   
#   So the goal here is that we get OOM data where processes survive long enough
#   send telemetry. To get the case where processes -crash- we'll also have to
#   get crash data. See Bug 1964092 for that half.
# ==============================================================================

diff -r 2f8d1dbbacf6 -r 5a65d2960e0d dom/base/use_counter_metrics.yaml
--- a/dom/base/use_counter_metrics.yaml	Thu May 15 19:59:25 2025 +0000
+++ b/dom/base/use_counter_metrics.yaml	Thu May 15 20:04:27 2025 +0000
@@ -107,8 +107,8 @@
     send_in_pings:
       - use-counters
 
-# Total of 2397 use counter metrics (excludes denominators).
-# Total of 358 'page' use counters.
+# Total of 2405 use counter metrics (excludes denominators).
+# Total of 362 'page' use counters.
 use.counter.page:
   svgsvgelement_getelementbyid:
     type: counter
@@ -824,6 +824,74 @@
     send_in_pings:
       - use-counters
 
+  js_large_oom_reported:
+    type: counter
+    description: >
+      Whether a page reported a large oom.
+      Compare against `use.counter.top_level_content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_small_oom_reported:
+    type: counter
+    description: >
+      Whether a page reported a small oom.
+      Compare against `use.counter.top_level_content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_large_oom_recovered:
+    type: counter
+    description: >
+      Whether a page recovered a large oom.
+      Compare against `use.counter.top_level_content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_small_oom_recovered:
+    type: counter
+    description: >
+      Whether a page recovered a small oom.
+      Compare against `use.counter.top_level_content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
   console_assert:
     type: counter
     description: >
@@ -6196,7 +6264,7 @@
     send_in_pings:
       - use-counters
 
-# Total of 358 'document' use counters.
+# Total of 362 'document' use counters.
 use.counter.doc:
   svgsvgelement_getelementbyid:
     type: counter
@@ -6912,6 +6980,74 @@
     send_in_pings:
       - use-counters
 
+  js_large_oom_reported:
+    type: counter
+    description: >
+      Whether a document reported a large oom.
+      Compare against `use.counter.content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_small_oom_reported:
+    type: counter
+    description: >
+      Whether a document reported a small oom.
+      Compare against `use.counter.content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_large_oom_recovered:
+    type: counter
+    description: >
+      Whether a document recovered a large oom.
+      Compare against `use.counter.content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
+  js_small_oom_recovered:
+    type: counter
+    description: >
+      Whether a document recovered a small oom.
+      Compare against `use.counter.content_documents_destroyed`
+      to calculate the rate.
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1852098
+    notification_emails:
+      - dom-core@mozilla.com
+      - emilio@mozilla.com
+    expires: never
+    send_in_pings:
+      - use-counters
+
   console_assert:
     type: counter
     description: >